src/app/notifications/notification-action.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * An action for a button, dropdown, etc:
+ *
+ * id - Optional unique Id for the filter field, useful for comparisons
+ * isDisabled - set to true to disable the action
+ * isSeparator - set to true if this is a placehodler for a separator rather than an action
+ * name - The name of the action, displayed on the button
+ * title - Optional title, used for the tooltip
+ */
+export class NotificationAction {
+  id?: string;
+  isDisabled?: boolean = false;
+  isSeparator?: boolean = false;
+  name: string;
+  title?: string;
+  callback: NotificationActionCallback;
```

Comment:
The Notification object in ngx-widgets supports Action objects with the same properties. The only addition here is a callback.

The notification service generates events when the user selects actions and close. The idea was to have the event also provides the original Action object you provided. If you have the action ID, name, title, etc., then you should have everything needed for comparison?

If you need an additional callback property, we can add it. I just didn't feel that it was absolutely necessary? We were trying to move away from calling people's callbacks ourselves in the component in favor of providing events.

If you need to duplicate all these classes on your own, then there is something missing from the Notification component. I'd rather make the necessary changes to the Notification component itself and support what you need. I'm guessing if we added a callback property, all these classes would be unnecessary?